tests/kola/networking/default-network-behavior-change/test.sh

```diff
@@ -0,0 +1,168 @@
+
+[ipv6]
+addr-gen-mode=eui64
+dhcp-timeout=90
+dns-search=
+method=auto
+
+[proxy]
+
+[user]
+org.freedesktop.NetworkManager.origin=nm-initrd-generator"
+# EXPECTED_INITRD_NETWORK_CFG2
+#   - used on all RHCOS releases
+#   - used on FCOS F34
+EXPECTED_INITRD_NETWORK_CFG2="[connection]
+id=Wired Connection
+uuid=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+type=ethernet
+autoconnect-retries=1
+multi-connect=3
+permissions=
+
+[ethernet]
+mac-address-blacklist=
+
+[ipv4]
+dhcp-timeout=90
+dns-search=
+method=auto
+
+[ipv6]
+addr-gen-mode=eui64
+dhcp-timeout=90
+dns-search=
+method=auto
+
+[proxy]"
+
+# EXPECTED_REALROOT_NETWORK_CFG1:
+#   - used on all FCOS/RHCOS releases
+EXPECTED_REALROOT_NETWORK_CFG1="[connection]
+id=Wired connection 1
+uuid=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+type=ethernet
+autoconnect-priority=-999
+interface-name=xxxx
+permissions=
+timestamp=xxxxxxxxxx
+
+[ethernet]
+mac-address-blacklist=
+
+[ipv4]
+dns-search=
+method=auto
+
+[ipv6]
+addr-gen-mode=stable-privacy
+dns-search=
+method=auto
+
+[proxy]
+
+[.nmmeta]
+nm-generated=true"
+
+# Function that will remove unique (per-run) data from a connection file
+normalize_connection_file() {
+    sed -e s/^uuid=.*$/uuid=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/ \
+        -e s/^timestamp=.*$/timestamp=xxxxxxxxxx/                 \
+        -e s/^interface-name=.*$/interface-name=xxxx/             \
+        $1
+}
+
+source /etc/os-release
+# All current FCOS releases use the same config
+if [ "$NAME" == "Fedora Linux" -o "$NAME" == "Fedora" ]; then
+    if [ "$VERSION_ID" -eq "34" ]; then
+        EXPECTED_INITRD_NETWORK_CFG=$EXPECTED_INITRD_NETWORK_CFG2
+        EXPECTED_REALROOT_NETWORK_CFG=$EXPECTED_REALROOT_NETWORK_CFG1
+    elif [ "$VERSION_ID" -ge "35" ]; then
+        EXPECTED_INITRD_NETWORK_CFG=$EXPECTED_INITRD_NETWORK_CFG1
+        EXPECTED_REALROOT_NETWORK_CFG=$EXPECTED_REALROOT_NETWORK_CFG1
+    else
+        fatal "fail: not operating on expected OS version"
+    fi
+elif [ "$NAME" == "Red Hat Enterprise Linux CoreOS" ]; then 
+    # For the version comparison use string substitution to remove the
+    # '.` from the version so we can use integer comparison
+    RHCOS_MINIMUM_VERSION="4.9"
+    if [ "${VERSION_ID/\./}" -ge "${RHCOS_MINIMUM_VERSION/\./}" ]; then
+        EXPECTED_INITRD_NETWORK_CFG=$EXPECTED_INITRD_NETWORK_CFG2
+        EXPECTED_REALROOT_NETWORK_CFG=$EXPECTED_REALROOT_NETWORK_CFG1
+    else
+        fatal "fail: not operating on expected OS version"
+    fi
+else
+    fatal "fail: not operating on expected OS"
+fi
+
+
+# Execute nm-initrd-generator against our default kargs (defined by
+# afterburn drop in) to get the generated initrd network config.
+DEFAULT_KARGS_FILE=/usr/lib/dracut/modules.d/35coreos-network/50-afterburn-network-kargs-default.conf 
+DEFAULT_KARGS=$(grep AFTERBURN_NETWORK_KARGS_DEFAULT $DEFAULT_KARGS_FILE | \
+                sed s/Environment=AFTERBURN_NETWORK_KARGS_DEFAULT=// |
+                tr -d "'" | tr -d '"')
```

Comment:
Ahh too bad. If this weren't a non-exclusive test, the service name would be predictable, so we could have a Butane config that copies the dropin to the external test.

Hmm, here's another approach:

```suggestion
source <(grep -o 'AFTERBURN_NETWORK_KARGS_DEFAULT=.*' $DEFAULT_KARGS_FILE)
```

Reply:
TIL about `grep -o`

Reply:
done